pkg/skaffold/runner/v1/deploy.go

```diff
@@ -41,28 +41,23 @@ import (
 
 // DeployAndLog deploys a list of already built artifacts and optionally show the logs.
 func (r *SkaffoldRunner) DeployAndLog(ctx context.Context, out io.Writer, artifacts []graph.Artifact) error {
-	// Update which images are logged.
-	r.AddTagsToPodSelector(artifacts)
-
-	logger := r.createLogger(out, artifacts)
-	defer logger.Stop()
+	defer r.deployer.GetLogger().Stop()
 
 	// Logs should be retrieved up to just before the deploy
-	logger.SetSince(time.Now())
+	r.deployer.GetLogger().SetSince(time.Now())
 	// First deploy
 	if err := r.Deploy(ctx, out, artifacts); err != nil {
 		return err
 	}
 
-	forwarderManager := r.createForwarder(out)
-	defer forwarderManager.Stop()
+	r.deployer.GetResourcePreviewer().Stop()
```

Comment:
should there be a `defer` here?

Reply:
oof yes there should. I wonder how I didn't catch that 🤔